### PR TITLE
Plushie hugs 

### PR DESCRIPTION
**Pull request by flrsh**
[FEATURE]

## About the PR 

When you attack someone with a plushie on help intent, the plushie will hug them.

## Why's this needed? 

It's cute.

## Changelog

```
(u)Flourish:
(+)Hit someone with a plush toy on help intent to make the plush toy hug them.
```

### CODE REVIEW

File: code/obj/item/toys.dm [hunk 1]
```diff
@@ -106,12 +106,18 @@
...
 		return
-	logTheThing("say", user, null, "makes [src] say,  \"[message]\"")
+	logTheThing("say", user, null, "makes [src] say, \"[message]\"")
 	user.audible_message("<span class='emote'>[src] says, \"[message]\"</span>")
 	var/mob/living/carbon/human/H = user
 	if (H.sims)
 		H.sims.affectMotive("fun", 1)
 
+/obj/item/toy/plush/attack(mob/M as mob, mob/user as mob)
+	if (user.a_intent == "help")
```
Comment (Sovexe) [inline code/obj/item/toys.dm:116]:
```suggestion
	if (user.a_intent == INTENT_HELP)
```

**Review by Sovexe — CHANGES_REQUESTED** [review]

File: code/obj/item/toys.dm (hunk 1 above)
Comment (flrsh) [reply code/obj/item/toys.dm]: Shoot I mean define not macro. Rip my unblemished commit history.

File: code/obj/item/toys.dm [hunk 2]
```diff
@@ -106,12 +106,18 @@
...
-	logTheThing("say", user, null, "makes [src] say,  \"[message]\"")
+	logTheThing("say", user, null, "makes [src] say, \"[message]\"")
 	user.audible_message("<span class='emote'>[src] says, \"[message]\"</span>")
 	var/mob/living/carbon/human/H = user
 	if (H.sims)
 		H.sims.affectMotive("fun", 1)
 
+/obj/item/toy/plush/attack(mob/M as mob, mob/user as mob)
+	if (user.a_intent == INTENT_HELP)
+		user.audible_message("<span class='emote'>[src] gives [M] a hug!</span>")
```
Comment (ZeWaka) [inline code/obj/item/toys.dm:117]: Possible to add a squeaky noise?

**Review by ZeWaka — APPROVED** [review]

File: code/obj/item/toys.dm (hunk 2 above)
Comment (ZeWaka) [inline code/obj/item/toys.dm:117]: Should be a visible message, not audible.

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/obj/item/toys.dm (hunk 2 above)
Comment (Sovexe) [inline code/obj/item/toys.dm:117]:
```suggestion
		user.visible_message("<span class='emote'>[src] gives [M] a hug!</span>")
```

Correct me if I'm wrong but I believe this should be `visible_message` or else deaf people will be banned from receiving hugs

Comment (flrsh) [reply code/obj/item/toys.dm:117]: If it's visible then blind people will be banned from receiving hugs!

Comment (flrsh) [reply code/obj/item/toys.dm:117]: No you.

Comment (Sartorius7) [reply code/obj/item/toys.dm:117]: Some sort of else condition outputting a message about feeling something hugging the target, perhaps?

**Review by Sovexe — APPROVED** [review]

**Review by ZeWaka — APPROVED** [review]

File: code/obj/item/toys.dm [hunk 3]
```diff
@@ -106,12 +106,18 @@
...
 	user.audible_message("<span class='emote'>[src] says, \"[message]\"</span>")
 	var/mob/living/carbon/human/H = user
 	if (H.sims)
 		H.sims.affectMotive("fun", 1)
 
+/obj/item/toy/plush/attack(mob/M as mob, mob/user as mob)
+	if (user.a_intent == INTENT_HELP)
+		M.visible_message("<span class='emote'>[src] gives [M] a hug!</span>", "<span class='emote'>[src] gives you a hug!</span>")
+	else
+		..()
```
Comment (ZeWaka) [inline code/obj/item/toys.dm:119]:
```suggestion
		. = ..()
```

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Review by ZeWaka — APPROVED** [review]

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**